### PR TITLE
PAYEE-761: Removes TaiReponse from updateEstimatedIncome in connector…

### PR DESCRIPTION
**Pull request by sergi-hmrc**
… TaxAccountConnector

- [ ] Unit tests passing
- [ ] Coverage reached
- [ ] Acceptance tests passing
- [ ] Reviewed

### CODE REVIEW

File: app/controllers/IncomeController.scala [hunk 1]
```diff
@@ -264,20 +261,15 @@ class IncomeController @Inject()(
...
-                incomeId.toInt) flatMap {
-                case TaiSuccessResponse =>
-                  estimatedPayJourneyCompletionService.journeyCompleted(incomeId).map { _ =>
-                    respondWithSuccess(incomeName, incomeId.toInt, incomeType, newAmount)
-                  }
-                case failure: TaiFailureResponse =>
-                  throw new RuntimeException(s"Failed to update estimated income with exception: ${failure.message}")
-              }
-            }
+            for {
```
Comment (PerrettJ4) [inline app/controllers/IncomeController.scala:264]: How come you even need to do this for-yield when all the values that are 'yielded' already exist in `val incomeName :: newAmount :: incomeId :: incomeType :: Nil = cache.toList`

**Review by PerrettJ4 — APPROVED** [review]
lgtm

File: app/uk/gov/hmrc/tai/connectors/TaxAccountConnector.scala [hunk 2]
```diff
@@ -117,13 +118,10 @@ class TaxAccountConnector @Inject()(httpHandler: HttpHandler, servicesConfig: Se
...
-    httpHandler.putToApi(updateTaxCodeIncome(nino.nino, year, id), UpdateTaxCodeIncomeRequest(newAmount)) map (_ =>
-      TaiSuccessResponse) recover {
-      case e: Exception =>
-        logger.warn(s"Error while updating estimated income for $nino with exception:${e.getMessage}")
-        TaiTaxAccountFailureResponse(e.getMessage)
-    }
+    implicit hc: HeaderCarrier): Future[Done] =
+    httpHandler
+      .putToApi(updateTaxCodeIncome(nino.nino, year, id), UpdateTaxCodeIncomeRequest(newAmount))
+      .map(_ => Done)
```
Comment (brendanstraughan) [inline app/uk/gov/hmrc/tai/connectors/TaxAccountConnector.scala:124]: do we need to add a recover on here?

Comment (sergi-hmrc) [reply app/uk/gov/hmrc/tai/connectors/TaxAccountConnector.scala:124]: we can but looking at where it was used... the recover was to wrap exceptions and then throw runtime ones to be caught again using NonFatal recover :D

Comment (brendanstraughan) [reply app/uk/gov/hmrc/tai/connectors/TaxAccountConnector.scala:124]: cool as long as it is caught higher up I am happy

**Review by brendanstraughan — APPROVED** [review]

**Pull request merged**